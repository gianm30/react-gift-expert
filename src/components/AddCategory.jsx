import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = evt => {
        evt.preventDefault()
        onNewCategory(inputValue);
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit}>
            <input type="text" placeholder="Buscar gifs" value={inputValue}
                onChange={onInputChange} />
        </form>
    );
}