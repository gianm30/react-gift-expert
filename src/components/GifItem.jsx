import PropTypes from 'prop-types';

export const GifItem = ({ image, id, title, url, onDeleteImage }) => {
    return (
        <div className="card">
            <div></div>
            <img src={url} alt={title} />
            <p key={id} data-testid="titulo">{title}<a onClick={evt => onDeleteImage(image)}>X</a></p>
        </div>
    )
}

GifItem.propTypes = {
    image: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    onDeleteImage: PropTypes.func.isRequired
}