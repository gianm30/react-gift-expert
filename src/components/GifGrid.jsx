import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import { PropTypes } from 'prop-types';


export const GifGrid = ({ category, onDeleteCategory }) => {
    const { images, setImages, isLoading } = useFetchGifs(category);
    const onDeleteImage = image => {
        const index = images.indexOf(image);
        console.log(index);
        if (index !== -1)
            setImages(images.filter(evt => evt !== image));
    }

    return (
        <>
            <h3>{category} <a onClick={evt => onDeleteCategory(category)}>X</a></h3>
            {
                isLoading && <h2>Cargando ...</h2>
            }

            <div className="card-grid">
                {
                    images.map(image => (
                        <GifItem key={image.id} image={image}
                            {...image}
                            onDeleteImage={onDeleteImage} />)
                    )
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    onDeleteCategory: PropTypes.func.isRequired
}