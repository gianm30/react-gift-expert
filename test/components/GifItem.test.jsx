import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe("Pruebas en <GifItem/>", () => {
    const id = "1";
    const image = {};
    const title = "Hola";
    const url = "https://pruebas-gian.com/carrasco.jpg";
    const onDeleteImage = () => { };
    const renderObj = <GifItem id={id} image={image} title={title} url={url} onDeleteImage={onDeleteImage} />;

    test("Debe de hacer match con el snapshot", () => {
        const { container } = render(renderObj);
        expect(container).toMatchSnapshot();
    })

    test("Debe mostrar la imagen con el URL y el ALT indicado", () => {
        render(renderObj);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test("Debe mostrar el tÃ­tulo en el componente", () => {
        render(renderObj);
        const { innerHTML } = screen.getByTestId('titulo');
        expect(innerHTML).toContain(title);
    })
})