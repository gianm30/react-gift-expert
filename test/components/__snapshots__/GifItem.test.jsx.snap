// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div />
    <img
      alt="Hola"
      src="https://pruebas-gian.com/carrasco.jpg"
    />
    <p
      data-testid="titulo"
    >
      Hola
      <a>
        X
      </a>
    </p>
  </div>
</div>
`;
